src/main/java/org/example/menu/GameShopMenu.java

```diff
@@ -0,0 +1,130 @@
+                case "5" -> walletMenu.recharge();
+                case "0" -> {
+                    isOn = false;
+                    userMenu.exit();
+                }
+            }
+        }
+    }
+
+    private String menuResponse() {
+        List<String> correctResponse = List.of("1", "2", "3", "4", "5", "0");
+        String menuResponse = scanner.next();
+        while (!checkResponse(menuResponse, correctResponse)) {
+            System.err.println(Messages.WRONG_OPTION);
+            menuResponse = scanner.next();
+        }
+        return menuResponse;
+    }
+
+    private boolean checkResponse(String gameResponse, List<String> correctResponse) {
+        return correctResponse.stream().
+                anyMatch(element -> element.equals(gameResponse));
+    }
+    private void showAllGames() {
+        gameService.showAll().forEach(System.out::println);
+    }
+
+    private void purchaseGameMenu() {
+        String gameResponse;
+        List<String> correctResponse = new ArrayList<>();
+        for (int i = 1; i <= gameService.getNumberOfGames(); i++) {
+            correctResponse.add(String.valueOf(i));
+        }
+        showAllGames();
+        System.out.println(Messages.CHOOSE_GAME_NUMBER);
+        gameResponse = scanner.next();
+        while (!checkResponse(gameResponse, correctResponse)) {
+            System.err.println(Messages.WRONG_OPTION);
+            gameResponse = scanner.next();
+        }
+        if (isEnoughMoney(gameResponse) && !isGameAlreadyBought(gameResponse)) {
+            System.out.println("Do you really want to buy " + gameService.findById(Integer.parseInt(gameResponse)).getName() + "?");
+            if (handleResponse()) purchaseGame(gameResponse);
+            else System.out.println(Messages.REJECT_GAME_PURCHASE);
+        } else if (!isEnoughMoney(gameResponse) && !isGameAlreadyBought(gameResponse)) {
+            System.err.println(Messages.NOT_ENOUGH_MONEY);
+            if (handleResponse()) walletMenu.recharge();
+            else System.out.println(Messages.REJECT_RECHARGE);
+        } else if ((isEnoughMoney(gameResponse) || !isEnoughMoney(gameResponse)) && isGameAlreadyBought(gameResponse)) {
+            System.err.println(Messages.GAME_ALREADY_BOUGHT);
+        }
+    }
+
+    private boolean isEnoughMoney(String gameResponse) {
+        return gameService.findById(Integer.parseInt(gameResponse)).getCost() <= walletMenu.showBalance();
+    }
+
+    private boolean isGameAlreadyBought(String gameResponse) {
+        List<Game> ownedGames = new ArrayList<>();
+        for (Integer id : userService.userGames(userSingleton.getUser().getId())) {
+            ownedGames.add(gameService.findById(id));
+        }
+        return ownedGames.stream().
+                anyMatch(game -> game.getName().equals(gameService.findByName(gameService.findById(Integer.parseInt(gameResponse)).getName())));
+    }
+
+    private boolean handleResponse() {
```

Comment:
it's better to use hasXXXX() or isXXXX() as a method name when return type is boolean